### ISSUE TITLE
Implement the domain server

### ISSUE THREAD

**Issue by goloroden** [thenativeweb/wolkenkit#342]
### What is this feature about?

In order to process commands and turn them into domain events, we need a domain server. When it receives a command, it needs to replay the event stream (maybe including snapshots) for the given aggregate, run the domain logic, and send the resulting domain events to the event store.

### What needs to be done to implement this feature?

- [x] Consume commands from the dispatcher
- [x] Handle commands and store events
  - [x] Create failed/rejected events
- [x] Acknowledge commands at dispatcher
- [x] Add tests

### What else should we know?

Commands may be received more than once, see #178 for details.

**Comment by goloroden** [comment]
As already discussed on Slack, I'm going to add a few notes here what we need to pay attention to (please note that this list is not complete):

- A domain server can process several commands in parallel. It should be configurable how many commands can be run in parallel. The dispatcher should ensure that they do not get in the way of each other.
- In the old `core` server, we put a lot of effort into authorization, but that should be much easier now. We just have to remember to call the `isAuthorized` function 😉 
- We can skip impersonation of commands for now. The existing implementation is pretty dumb and not well-thought. We should talk about this separately. I'll create another issue and link it to this one.
- It is important to trigger the `...Failed` and `...Rejected` domain events in the different error cases.
